src/chk/chk_leader.c

```diff
@@ -2264,6 +2266,17 @@ chk_leader_sched(void *args)
 		if (more_dead)
 			goto check_dead;
 
+		if (!d_list_empty(&ins->ci_interaction_filter_list)) {
+			pending = d_list_pop_entry(&ins->ci_interaction_filter_list,
+						   struct chk_pending_rec, cpr_ins_link);
+			act     = ins->ci_prop.cp_policies[pending->cpr_class];
+			if (pending->cpr_action != CHK__CHECK_INCONSIST_ACTION__CIA_INTERACT ||
+			    !chk_is_valid_action(pending, act))
+				d_list_add_tail(&pending->cpr_ins_link, &ins->ci_pending_list);
+			else
+				chk_leader_act(pending->cpr_seq, act);
+		}
```

Comment:
I'm a bit less familiar with the internals of the checker on the C side. Why do we only address one pending item from the list, rather than iterating through all of them?

Reply:
This part is inside `chk_leader_sched()` logic that is a `while()` blob(). For each cycle it needs to handle many things, including the pending action. So each time, it only handles the top pending action on the stack (or in the list), next cycle will be the next one.